templates/product/details.html

```diff
@@ -161,7 +161,7 @@ <h2 class="product__info__price">
           </h2>
         {% endif %}
       {% endif %}
-      {% if is_visible and product.is_in_stock %}
+      {% if is_visible and product.is_in_stock and availability.available %}
```

Comment:
If you look at how `availability.available` works, it actually covers both `is_visible` check and `product.is_in_stock`, so we could simple have:

```suggestion
      {% if availability.available %}
```

Reply:
Yea, true.

Reply:
So if you agree, please commit this suggestion, update the `CHANGELOG.md` and it will be good to merge. Thanks!